presidio-image-redactor/presidio_image_redactor/dicom_image_pii_verify_engine.py

```diff
@@ -0,0 +1,336 @@
+        :param kwargs: Additional values for the analyze method in ImageAnalyzerEngine.
+        :return: Evaluation comparing redactor engine results vs ground truth.
+        """
+        # Verify detected PHI
+        verify_image, ocr_results, analyzer_results = self.verify_dicom_instance(
+            instance, padding_width, **kwargs
+        )
+        formatted_ocr_results = self._get_bboxes_from_ocr_results(ocr_results)
+        detected_phi = self._get_bboxes_from_analyzer_results(analyzer_results)
+
+        # Remove duplicate entities in results
+        detected_phi = self._remove_duplicate_entities(detected_phi)
+
+        # Get correct PHI text (all TP and FP)
+        all_pos = self._label_all_positives(
+            ground_truth, formatted_ocr_results, detected_phi, tolerance
+        )
+
+        # Calculate evaluation metrics
+        precision = self._calculate_precision(ground_truth, all_pos)
+        recall = self._calculate_recall(ground_truth, all_pos)
+
+        eval_results = {
+            "all_positives": all_pos,
+            "ground_truth": ground_truth,
+            "precision": precision,
+            "recall": recall,
+        }
+
+        return verify_image, eval_results
+
+    @staticmethod
+    def _get_bboxes_from_ocr_results(ocr_results: dict) -> dict:
+        """Get bounding boxes on padded image for all detected words from ocr_results.
+
+        Args:
+            ocr_results (dict): Results from OCR.
+
+        Return:
+            bboxes (dict): Bounding box information per word.
+        """
+        bboxes = {}
+        item_count = 0
+        for i in range(0, len(ocr_results["text"])):
+            detected_text = ocr_results["text"][i]
+            if len(detected_text) > 0:
+                bboxes[str(item_count)] = {
+                    "left": ocr_results["left"][i],
+                    "top": ocr_results["top"][i],
+                    "width": ocr_results["width"][i],
+                    "height": ocr_results["height"][i],
+                    "conf": float(ocr_results["conf"][i]),
+                    "label": detected_text,
+                }
+                item_count += 1
+
+        return bboxes
+
+    @staticmethod
+    def _remove_duplicate_entities(results: dict, dup_pix_tolerance: int = 5) -> dict:
+        """Handle when a word is detected multiple times as different types of entities.
+
+        Args:
+            results (dict): Detected PHI.
+            dup_pix_tolerance (int): Pixel difference tolerance for
+            identifying duplicates.
+
+        Return:
+            results_no_dups (dict): Detected PHI with no duplicate entities.
+        """
+        dups = []
+        results_no_dups = results.copy()
+
+        # Check for duplicates
+        for i in results:
+            i_left = results[i]["left"]
+            i_top = results[i]["top"]
+            i_width = results[i]["width"]
+            i_height = results[i]["height"]
```

Comment:
I am not familiar with how to use dunders in this case. Would you mind sharing an example and/or resource?

Reply:
In case we want to check for BBs equality or that one is contained within the other, we can implement the `__equals__` and `__container__` methods respectfully.

Reply:
The left, top, width, and height attributes are available in the returned list.

I have decided to use a dictionary approach to reduce the lines of code for now. I will revisit the use of dunders in the future PR for moving the bounding box code to a common module.